src/environment/EnvironmentManager.py

```diff
@@ -85,8 +89,38 @@ def run(env_name: str, *args, **kwargs) -> int:
         print(f"======== {agent_file} ========")
         res = system(
             f"conda run -n {EnvironmentManager.CONDA_ENV_PREFIX}_{env_name} "
+            + "--no-capture-output "
             + f"python {agent_file} {args_str} {kwargs_str}"
         )
         print(f"========={'=' * len(str(agent_file))}=========")
 
         return res == 0
+
+    @staticmethod
+    def send_latent_codes(latent_codes: np.ndarray) -> None:
+        """
+        Server sending latent codes via localhost 6969 port.
+
+        Args:
+            latent_codes (np.ndarray): Latent codes to send.
+        """
+        threading.Thread(target=_latent_code_agent, args=(latent_codes,)).start()
+
+
+def _latent_code_agent(latent_codes: np.ndarray) -> None:
```

Comment:
maybe space contents a bit more logically, I would personally add a couple of more comments as well